string_calculator.rb

```diff
@@ -0,0 +1,42 @@
+class StringCalculator
+  def add(string_of_numbers = "")
+    sum = 0
+    separate_string = string_of_numbers.split(",")
+    numbers = separate_string.collect do |n|
+      n.to_i
+    end
+    neg = numbers.select(&:negative?)
+    is_negative = numbers.any? { |x| x.negative? }
+    is_greater_than_1000 = numbers.any? { |x| x >= 1000 }
```

Comment:
Instead of doing this check, you can just build a list of `numbers` that has anything greater than 1000 excluded. So something like this would work too:

`numbers = numbers.reject { |number| number > 1000 }`. Also avoid using short variable names like `x` and `n` since they don't communicate what they are and hinder readability.